### ISSUE TITLE
Jest tests fail with Babel error

### ISSUE THREAD

**Issue by johnwatkins0** [10up/10up-toolkit#69]
**Describe the bug**

Working off of https://github.com/10up/block-components/pull/30 on the block-components library, I was attempting to add a Jest test and found the tests couldn't run due to a Babel error that I believe is coming from somewhere inside the dependencies, potentially from 10up-toolkit. See the steps to reproduce.

```
> @10up/block-components@1.1.10 test
> 10up-toolkit test-unit-jest

 FAIL  components/__tests__/index.js
  ● Test suite failed to run

    [BABEL] /Users/johnwatkins/jw-block-components/components/__tests__/index.js: Unknown option: .wordpress. Check out https://babeljs.io/docs/en/babel-core/#options for more information about options.
    - Maybe you meant to use
    "preset": [
      ["@10up/babel-preset-default", {
      "wordpress": true
    }]
    ]
    To be a valid preset, its name and options should be wrapped in a pair of brackets

      at throwUnknownError (node_modules/@babel/core/lib/config/validation/options.js:133:27)
      at node_modules/@babel/core/lib/config/validation/options.js:118:5
          at Array.forEach (<anonymous>)
      at validateNested (node_modules/@babel/core/lib/config/validation/options.js:94:21)
      at validate (node_modules/@babel/core/lib/config/validation/options.js:85:10)
      at node_modules/@babel/core/lib/config/full.js:324:36
      at cachedFunction (node_modules/@babel/core/lib/config/caching.js:60:27)
          at cachedFunction.next (<anonymous>)
      at evaluateSync (node_modules/gensync/index.js:251:28)
      at sync (node_modules/gensync/index.js:89:14)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        2.358 s
Ran all test suites.
```

Additionally, the ESLint config that comes from toolkit doesn't seem to bring the Jest config, so I get linting issues from my test file. I suggest we either include that in the 10up config or provide guidance in this repo on how to include. 

<img width="444" alt="Screen Shot 2021-06-03 at 8 48 33 AM" src="https://user-images.githubusercontent.com/9020968/120655638-80a63e00-c448-11eb-8a24-45e963e1d73c.png">

**Steps to Reproduce**

1. Clone https://github.com/johnwatkins0/block-components
2. Checkout `fail-test-demo`
3. Run `npm install`
4. Run `npm test`

**Comment by nicholasio** [comment]
The jest issue is bc of this line: https://github.com/10up/10up-toolkit/blob/f3069b0e88e818ac83728e31a78f9c54a5980742/packages/toolkit/config/babel-transform.js#L7

It should be `	presets: [['@10up/babel-preset-default', { wordpress: true }]],` instead. You should be able to fix in the meantime this by overriding the jest babel config and making the fix above.

And as far as jest linting you need to extend `@10up/eslint-config/jest` to enable those rules.

**Comment by nicholasio** [comment]
This should be fixed in 10up-toolkit@1.0.7

**Comment by johnwatkins0** [comment]
Thanks, @nicholasio!
